backend/estimate-service/src/main/java/org/egov/service/NotificationService.java

```diff
@@ -0,0 +1,262 @@
+                EstimateServiceConstant.ESTIMATE_NOTIFICATION_ENG_LOCALE_CODE, EstimateServiceConstant.ESTIMATE_MODULE_CODE);
+        return localizedMessageMap.get(EstimateServiceConstant.ESTIMATE_NOTIFICATION_ENG_LOCALE_CODE + "|" + rootTenantId).get(msgCode);
+    }
+
+    /**
+     * Builds msg based on the format
+     *
+     * @param estimate
+     * @param message
+     * @param userDetailsForSMS
+     * @return
+     */
+    public String buildMessageForRejectAction(Estimate estimate, Map<String, String> userDetailsForSMS, String message) {
+        message = message.replace("{estimate_no}", estimate.getEstimateNumber())
+                .replace("{project_name}", userDetailsForSMS.get("projectName"))
+                .replace("{location}", userDetailsForSMS.get("locationName"))
+                .replace("{username}", userDetailsForSMS.get("userName"))
+                .replace("{designation}", userDetailsForSMS.get("designation"));
+        return message;
+    }
+
+    public String buildMessageForApproveAction_Creator(Estimate estimate, Map<String, String> userDetailsForSMS, String message) {
+        message = message.replace("{estimate_no}", estimate.getEstimateNumber())
+                .replace("{project_name}", userDetailsForSMS.get("projectName"))
+                .replace("{location}", userDetailsForSMS.get("locationName"));
+        return message;
+    }
+
+    /**
+     * Creates a cache for localization that gets refreshed at every call.
+     *
+     * @param requestInfo
+     * @param rootTenantId
+     * @param locale
+     * @param module
+     * @return
+     */
+    public Map<String, Map<String, String>> getLocalisedMessages(RequestInfo requestInfo, String rootTenantId, String locale, String module) {
+        Map<String, Map<String, String>> localizedMessageMap = new HashMap<>();
+        Map<String, String> mapOfCodesAndMessages = new HashMap<>();
+        StringBuilder uri = new StringBuilder();
+        RequestInfoWrapper requestInfoWrapper = new RequestInfoWrapper();
+        requestInfoWrapper.setRequestInfo(requestInfo);
+        uri.append(config.getLocalizationHost()).append(config.getLocalizationContextPath())
+                .append(config.getLocalizationSearchEndpoint()).append("?tenantId=" + rootTenantId)
+                .append("&module=" + module).append("&locale=" + locale);
+        List<String> codes = null;
+        List<String> messages = null;
+        Object result = null;
+        try {
+            result = repository.fetchResult(uri, requestInfoWrapper);
+            codes = JsonPath.read(result, EstimateServiceConstant.ESTIMATE_LOCALIZATION_CODES_JSONPATH);
+            messages = JsonPath.read(result, EstimateServiceConstant.ESTIMATE_LOCALIZATION_MSGS_JSONPATH);
+        } catch (Exception e) {
+            log.error("Exception while fetching from localization: " + e);
+        }
+        if (null != result) {
```

Comment:
I think result != null

Reply:
this is a more accurate way. This prevents nullPointer Exception